mmdet3d/datasets/pipelines/transforms_3d.py

```diff
@@ -740,8 +746,20 @@ def points_random_sampling(self,
         """
         if replace is None:
             replace = (points.shape[0] < num_samples)
-        choices = np.random.choice(
-            points.shape[0], num_samples, replace=replace)
+        sample_range = range(len(points))
+        if dist_metric:
+            depth = points.tensor[:, 2].numpy()
```

Comment:
We should use the API in the points structure to find the points to be kept. Thus this API can be applied to all coordinates.